File: P1/Ejercicio4/filters/email_text_filter.rb
# filtro que valida que el correo tenga texto antes del carácter '@'
# implementa la interfaz Filter
class EmailTextFilter
  include Filter

  def execute(request)
    email = request[:email]

    # separa la parte local del correo (antes del '@')
    local_part = email.split("@")[0]

    # comprueba que exista texto antes del '@'
    if local_part.nil? || local_part.empty?
      raise ValidationError, "El correo debe tener texto antes del @"
    end
  end
end

File: P2/Ejercicio1_MantenimientoFlutter/lib/filters/EmailTextFilter.dart
import '../interfaces/Filter.dart';

// filtro que valida el formato del email
class EmailTextFilter implements Filter {
  @override
  String? execute(Map<String, String> request) {
    final email = request['email'] ?? '';
    final parts = email.split('@');
    
    if (parts[0].isEmpty) {
      return "El correo debe tener texto antes del @";
    }
    return null;
  }
}